Assets/BossRoom/Scripts/Client/Net/ClientGameNetPortal.cs

```diff
@@ -211,6 +218,91 @@ public static bool StartClientRelayMode(GameNetPortal portal, string roomKey, ou
             return true;
         }
 
+        public static bool StartClientUnityRelayMode(GameNetPortal portal, string joinCode, out string failMessage)
+        {
+
+            var chosenTransport  = NetworkManager.Singleton.gameObject.GetComponent<TransportPicker>().UnityRelayTransport;
+            NetworkManager.Singleton.NetworkConfig.NetworkTransport = chosenTransport;
+
+            switch (chosenTransport)
+            {
+                case UnityTransport utp:
+                    Debug.Log($"Setting Unity Relay client with join code {joinCode}");
+                    Task t = Task.Run( () => {UnityServices.InitializeAsync();});
+                    if (!t.Wait(30000))
+                    {
+                        Debug.Log("UnityServices.Initialize did not finish within 30 seconds");
+                    }
+                    Debug.Log(AuthenticationService.Instance);
+                    if (!AuthenticationService.Instance.IsSignedIn)
+                    {
+                        Task t2 = Task.Run( () => {AuthenticationService.Instance.SignInAnonymouslyAsync();});
+                        t2.Wait(30000);
+                        var playerId = AuthenticationService.Instance.PlayerId;
+                        Debug.Log(playerId);
+
+                        var clientRelayUtilityTask = RelayUtility.JoinRelayServerFromJoinCode(joinCode);

```

Comment:
we only join at first signin?

Reply:
I mean you have to be signed in order to use relay ?

Reply:
what I mean is Boss Room has a full flow: when you finish the boss or you die, you're brought back to the main menu where you can launch a new game.
With `if (!AuthenticationService.Instance.IsSignedIn)`, that means if you're already signed in, you can't Join, it'll just skip this whole code block.

Reply:
That code is not used. Hence, I removed it